sql27thsep/sql27th sep.txt

```diff
@@ -0,0 +1,224 @@
+
+20. From a 'Products' table, write a query to replace all spaces in product names with underscores.
+select replace(productname,' ','-') from products;
+
+21. Create a string of 10 asterisks (*).
+select replicate('*',10);
+
+22. Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+select right(replicate('0',10)+[product code],10) from products;
+
+23. Insert the string 'New ' at the beginning of 'York City'.
+select concat_ws(' ','new'+ 'York city');    
+
+24. From an 'Emails' table, mask the username part of email addresses, showing only the first and last characters.
+select concat_ws('',left(emailID,1),replicate('*',charindex('@',emailID)),right(emailID,len(emailID)-charindex('@',emailID)+2)) from employees;
+
+25. Convert the string 'sql server' to uppercase.
+select upper('sql server');
+
+26. Write a query to convert all customer names in a 'Customers' table to uppercase.
+select upper(customername) from customers;
+
+27. Convert the string 'SQL SERVER' to lowercase.
+select lower('SQL SERVER');
+
+28. From a 'Products' table, write a query to convert all product descriptions to lowercase.
+select lower(productdescription) from products;
+
+29. Remove trailing spaces from the string 'SQL Server    '.
+select rtrim('SQL Server    ', ' ');
+
+30. Write a query to remove trailing spaces from all email addresses in an 'Employees' table.
+select rtrim(emailID,' ') from employees;
+
+31. Remove leading spaces from the string '   SQL Server'.
+select ltrim('   SQL Server',' ');
+
+32. From a 'Comments' table, write a query to remove leading spaces from all comment texts.
+select ltrim(comments,' ') from comments;
+
+33. Display the current date in the format 'dd-MM-yyyy'.
+select format(getdate(),'dd-MM-yyyy');
+
+
+34. From an 'Orders' table with an 'OrderTotal' column, display the total as a currency with 2 decimal places.
+select round(OrderTotal,2) from orders;
+
+35. Separate the string 'apple,banana,cherry' into individual fruits.
+declare @fruits nvarchar(30)='apple,banana,cherry';
+select value as fruitName from string_split(@fruits,',');
+
+36. From a 'Skills' table with a 'SkillList' column containing comma-separated skills, write a query to create a row for each individual skill.
+select value as individualSkill from skills cross apply string_split(skillList,',');
+
+
+Date and Time Functions
+
+37. Write a query to display the current date and time.
+select GETDATE();
+
+38. From an 'Orders' table, find all orders placed in the last 24 hours.
+select * from orders where DATEDIFF(hour,orderdate,getdate())<=24;
+
+39. Display the current UTC date and time.
+select SYSUTCDATETIME();
+
+40. Write a query to show the time difference between local time and UTC time.
+select datediff(hour,sysdatetime(),sysutcdatetime());
+
+
+41. Convert the current date and time to 'Pacific Standard Time'.
+select sysdatetime() at time zone 'pacific standard time' as pacific_time;
```

Comment:
(https://learn.microsoft.com/en-us/sql/t-sql/queries/at-time-zone-transact-sql?view=sql-server-ver16#b-convert-values-between-different-time-zones)

First we need to specify the time zone of our stored data, then specify the target time zone.